filebeat/harvester/reader/multiline.go

```diff
@@ -112,8 +114,18 @@ func (mlr *Multiline) Next() (Message, error) {
 	return mlr.state(mlr)
 }
 
+func (r *Multiline) Close() error {
+	close(r.done)
+	return r.reader.Close()
+}
+
 func (mlr *Multiline) readFirst() (Message, error) {
 	for {
+		select {
+		case <-mlr.done:
+			return Message{}, ErrReaderStopped
+		default:
+		}
```

Comment:
You are right, the channel should get obsolete as Next() will return an error anyways.

About your second comment: Agree. Will have a look, but potentially in a second PR to make sure this PR has in general no "functional" changes. In case we break something with the multiline change above, it is easier to find.